cmd/minikube/cmd/docker-env.go

```diff
@@ -250,7 +251,7 @@ var dockerEnvCmd = &cobra.Command{
 			}
 		}
 		sh := shell.EnvConfig{
-			Shell: shl,
+			Shell: strings.TrimSuffix(shl, filepath.Ext(shl)),
```

Comment:
can u add a comment what u is Before the Trim ?
and maybe we could limit doing this only on windows and powershell so we don't do un-intentional side effect for others ?

Reply:
According to my experiment, if I run `minikube docker-env` through ssh, the shl returned is *powershell.exe*, but if I run the same command on directly on windows, the shl is assigned to *powershell*.

**SSH on windows**
```
PS >.\minikube-windows-amd64.exe docker-env -v=8 --alsologtostderr
I0303 23:50:26.148593   17808 out.go:229] Setting OutFile to fd 280 ...
I0303 23:50:26.150141   17808 out.go:281] isatty.IsTerminal(280) = true
I0303 23:50:26.150141   17808 out.go:242] Setting ErrFile to fd 284...
I0303 23:50:26.150141   17808 out.go:281] isatty.IsTerminal(284) = true
W0303 23:50:26.159300   17808 docker-env.go:248] !!!!!!!!!!!! powershell.exe
I0303 23:50:26.160422   17808 mustload.go:66] Loading cluster: minikube
```
I think this caused the problem.

Reply:
Would be also nice to add a unit test to verify correct processing on different platforms

Reply:
@ilya-zuyev But this case is not easy to add a unit test. This *shl* is set by `shell.Detect()` and ultimately it is set by the libmachine API. And this function call resides in an anonymous function defined for Combra,  I am not sure how to add a unit test case for this?

```
Run: func(cmd *cobra.Command, args []string) {
	var err error
	shl := shell.ForceShell
	if shl == "" {
		shl, err = shell.Detect()
		if err != nil {
			exit.Error(reason.InternalShellDetect, "Error detecting shell", err)
		}
	}
	sh := shell.EnvConfig{
		Shell: strings.TrimSuffix(shl, filepath.Ext(shl)),
	}
```
Therefore, test it in integration test may be a better way?
Besides, do we really need a unit test here? the integration test itself, if passes, could already verify its correctness?

Reply:
@hetong07 feel free to make a PR to libmachine, we have our own fork once u improve this in libmachine, it will be better
here is link to the fork
https://github.com/machine-drivers/machine

once made the PR there plz ping me or @afbjorklund to merge it

Reply:
Created this PR https://github.com/machine-drivers/machine/pull/33.